### ISSUE TITLE
Add red/green highlight for diff syntax

### ISSUE THREAD

**Issue by mraible** [asciidoctor/asciidoctor-pdf#345]
I'm using [source,diff] to highlight a code block that's written in Java. It'd be great to allow both java and diff, but in the meantime, it'd be great if the PDF version looked as good as the HTML5 version.

PDF version:
<img width="730" alt="pdf-diff" src="https://cloud.githubusercontent.com/assets/17892/10790322/3a35c01e-7d48-11e5-86c9-bb527edc5790.png">

HTML5 version:
<img width="998" alt="html5-diff" src="https://cloud.githubusercontent.com/assets/17892/10790329/40a3187a-7d48-11e5-83f5-32c668530810.png">

**Comment by mojavelinux** [comment]
By addressing #306, we get this fix for free. Currently, we aren't implementing background colors for spans. Prawn does support the means for doing this, but it doesn't come "for free". It's necessary to use the `render_behind` hook to lay down the background color over the spanned region, then Prawn follows up with laying down the text.

**Comment by mojavelinux** [comment]
This shortcoming also prevents individual lines from being highlighted. For example:

```asciidoc
[source,python,highlight=4]
----
import re
for test_string in ['555-1212', 'ILL-EGAL']:
    if re.match(r'^\d{3}-\d{4}$', test_string):
        print test_string, 'is a valid US local phone number'
    else:
        print test_string, 'rejected'
----
```

**Comment by mojavelinux** [comment]
This was resolved by #663 for rouge. If you want the background color to stretch the length of the line, you need to create a custom rouge theme. See https://github.com/asciidoctor/asciidoctor-pdf/blob/master/docs/theming-guide.adoc#source-highlighting-theme for details.

I have not yet decided whether to implement this feature for Pygments. Pygments converts to HTML, which is then parsed by the text formatter. The text formatter would need to be enhanced to recognize the background color style.
